src/pudl/analysis/allocate_net_gen.py

```diff
@@ -614,18 +614,20 @@ def associate_generator_tables(
             on=IDX_GENS,
             how="outer",
         )
-        .merge(
-            gf,
-            on=IDX_PM_ESC,
-            how="outer",
-            validate="m:1",
-        )
+        .merge(gf, on=IDX_PM_ESC, how="outer", validate="m:1", indicator=True)
         .pipe(remove_inactive_generators)
+        .pipe(
```

Comment:
non-blocking: I remember you mentioning that some weird data slipped through without this. What was going on, and why did double-applying the `_allocate_unassociated_pm_records` method fix it?